sotodlib/io/bookbinder.py

```diff
@@ -508,13 +619,15 @@ def __call__(self, f):
             List containing frames to be entered into output Book
         """
 
-        def generate_missing_smurf_samples(t, dtype=np.int64):
+        def generate_missing_smurf_samples(f, t):
```

Comment:
If you're working on this function then wouldn't this be the place to fill in missing samples / make the number of samples per UFM all line up? (https://github.com/simonsobs/sotodlib/issues/389) 

It's very inter-related to this PR and is equally going to prevent us from having a working loader function / run analysis off books. So if it's not this PR it needs to be the priority for the next one.

Reply:
The current gap checker (this function) only checks for gaps only between—not within—frames. I have a good idea of how I can implement the latter, but I think it would be a large enough operation that it should be its own PR.